service/kas/access/rewrap.go

```diff
@@ -141,16 +167,55 @@ func verifySignedRequestToken(ctx context.Context, in *kaspb.RewrapRequest) (*Re
 	switch publicKey := clientPublicKey.(type) {
 	case *rsa.PublicKey:
 		requestBody.PublicKey = publicKey
-		return requestBody, nil
+		return &requestBody, nil
 	case *ecdsa.PublicKey:
 		requestBody.PublicKey = publicKey
-		return requestBody, nil
+		return &requestBody, nil
 	default:
 		slog.WarnContext(ctx, fmt.Sprintf("clientPublicKey not a supported key, was [%T]", clientPublicKey))
 		return nil, err400("clientPublicKey unsupported type")
 	}
 }
 
+func verifySRT(ctx context.Context, srt string, dpopJWK jwk.Key) (string, error) {
+	v, err := jws.NewVerifier(jwa.RS256)
```

Comment:
@dmihalcik-virtru  I might be missing something but why did you move away from `jwt.Parse` ? When passing in `jwt.WithKey` it should verify the signature.

Reply:
This is a JWS not a JWT.

Reply:
In our case they are pretty much the same. Our payload is JSON and since we are verifying using a signature algorithm our JWS will always be a JWT.

Reply:
yeah I guess we actually are doing JWT stuff - notably, we support setting the expiry. I've added it back